From ba2cb820f32994482a1918bd16bd0a548bb394ac Mon Sep 17 00:00:00 2001
From: Craig Andrews <candrews@integralblue.com>
Date: Tue, 12 Jun 2018 12:28:41 -0400
Subject: [PATCH] [dns-cache] Only use nmblookup if HAS_FILESYSTEM_SMB

nmblookup is part of SmbClient so only use it if SmbClient is present.

Fixes https://trac.kodi.tv/ticket/17916
---
 xbmc/network/DNSNameCache.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xbmc/network/DNSNameCache.cpp b/xbmc/network/DNSNameCache.cpp
index e993c536bf41..b30c60baf738 100644
--- a/xbmc/network/DNSNameCache.cpp
+++ b/xbmc/network/DNSNameCache.cpp
@@ -54,7 +54,7 @@ bool CDNSNameCache::Lookup(const std::string& strHostName, std::string& strIpAdd
   if(g_DNSCache.GetCached(strHostName, strIpAddress))
     return true;
 
-#ifndef TARGET_WINDOWS
+#if !defined(TARGET_WINDOWS) && defined(HAS_FILESYSTEM_SMB)
   // perform netbios lookup (win32 is handling this via gethostbyname)
   char nmb_ip[100];
   char line[200];
